kind/resource_cluster_test.go

```diff
@@ -0,0 +1,272 @@
+nodes:
+- role: control-plane
+- role: worker
+`),
+				),
+			},
+			{
+				Config: testAccBasicWaitForReadyExtraConfigClusterConfig(clusterName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccSimpleCluster(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "name", clusterName),
+					resource.TestCheckResourceAttr(resourceName, "node_image", kindDefaults.Image),
+					resource.TestCheckResourceAttr(resourceName, "wait_for_ready", "true"),
+					resource.TestCheckResourceAttr(resourceName, "kind_config", `kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+- role: worker
+`),
+				),
+			},
+			{
+				Config: testAccNodeImageClusterConfig(clusterName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccSimpleCluster(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "name", clusterName),
+					resource.TestCheckResourceAttr(resourceName, "node_image", nodeImage),
+					resource.TestCheckResourceAttr(resourceName, "wait_for_ready", "false"),
+					resource.TestCheckNoResourceAttr(resourceName, "kind_config"),
+				),
+			},
+			{
+				Config: testAccNodeImageWaitForReadyClusterConfig(clusterName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccSimpleCluster(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "name", clusterName),
+					resource.TestCheckResourceAttr(resourceName, "node_image", nodeImage),
+					resource.TestCheckResourceAttr(resourceName, "wait_for_ready", "true"),
+					resource.TestCheckNoResourceAttr(resourceName, "kind_config"),
+				),
+			},
+			{
+				Config: testAccNodeImageExtraConfigClusterConfig(clusterName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccSimpleCluster(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "name", clusterName),
+					resource.TestCheckResourceAttr(resourceName, "node_image", nodeImage),
+					resource.TestCheckResourceAttr(resourceName, "wait_for_ready", "false"),
+					resource.TestCheckResourceAttr(resourceName, "kind_config", `kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+- role: worker
+`),
+				),
+			},
+			{
+				Config: testAccNodeImageWaitForReadyExtraConfigClusterConfig(clusterName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccSimpleCluster(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "name", clusterName),
+					resource.TestCheckResourceAttr(resourceName, "node_image", nodeImage),
+					resource.TestCheckResourceAttr(resourceName, "wait_for_ready", "true"),
+					resource.TestCheckResourceAttr(resourceName, "kind_config", `kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+- role: worker
+`),
+				),
+			},
+			// TODO: add this for when resource update is implemented
+			// {
+			// 	ResourceName:      resourceName,
+			// 	ImportState:       true,
+			// 	ImportStateVerify: true,
+			// },
+		},
+	})
+}
+
+// TODO: check if this makes sense for kind
```

Comment:
What you could do here is use [provider.List](https://github.com/kubernetes-sigs/kind/blob/master/pkg/cluster/provider.go#L152) and check that the cluster name is not in the list.

Reply:
Good idea, added this suggestion.